app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt

```diff
@@ -121,8 +122,22 @@ class ThreeDotMenuMainRobot {
     }
 
     fun verifyAddonAvailableInMainMenu(addonName: String) {
-        onView(withText(addonName))
-            .check(matches(isDisplayed()))
+        try {
+            assertTrue(
+                "Addon not listed in the Add-ons menu",
+                mDevice.findObject(UiSelector().text(addonName)).waitForExists(waitingTime)
+            )
+        } catch (e: AssertionError) {
+            mDevice.pressBack()
+            browserScreen {
+            }.openThreeDotMenu {
+                openAddonsSubList()
+                assertTrue(
+                    "Addon not listed in the Add-ons menu",
+                    mDevice.findObject(UiSelector().text(addonName)).waitForExists(waitingTime)
+                )
```

Comment:
I'd like to get into the habit of using this model for retry rather than reduplicating code. We should add a retryCount global to constants now that I'm thinking about it, maybe set it to 3.  This will be helpful for individual checks.

https://github.com/mozilla-mobile/fenix/blob/main/app/src/androidTest/java/org/mozilla/fenix/helpers/RetryTestRule.kt#L21

e.g,
```Kotlin
for (i in 1..retryCount) {
  try {
   assertTrue()
   break
} catch (e: AssertionError) {
  if (I == retryCount) {
   throw t  
 } else {
   // actions, such as go back, click again
}
}
```
What do you think?

Reply:
Sure, I will try that

Reply:
done.